src/jenkins.py

```diff
@@ -288,47 +278,94 @@ def _unlock_wizard(container: ops.Container) -> None:
+            make_dirs=True,
+            user=USER,
+            group=GROUP,
+        )
+        container.push(
+            WIZARD_VERSION_PATH,
+            version,
+            encoding="utf-8",
+            make_dirs=True,
+            user=USER,
+            group=GROUP,
+        )
+    except (ops.pebble.PathError, JenkinsError) as exc:
+        raise JenkinsBootstrapError("Failed to unlock wizard.") from exc
 
 
-def _install_configs(container: ops.Container) -> None:
+def _install_config(container: ops.Container, filename: str, destination_path: Path) -> None:
     """Install jenkins-config.xml.
 
+    Args:
+        container: The Jenkins workload container.
+        filename: the source file to copy contents from.
+        destination_path: the target path.
+
+    Raises:
+        JenkinsBootstrapError: if the config can not be installed.
+
+    """
+    try:
+        with open(filename, encoding="utf-8") as jenkins_config_file:
+            container.push(destination_path, jenkins_config_file, user=USER, group=GROUP)
+    except ops.pebble.PathError as exc:
+        raise JenkinsBootstrapError("Failed to install configuration.") from exc
+
+
+def _install_configs(container: ops.Container) -> None:
+    """Install jenkins-config.xml and logging files.
+
+    Args:
+        container: The Jenkins workload container.
+    """
+    install_default_config(container)
+    _install_config(container, JENKINS_LOGGING_CONFIG, LOGGING_CONFIG_PATH)
+
+
+def install_default_config(container: ops.Container) -> None:
+    """Install default jenkins-config.xml.
+
     Args:
         container: The Jenkins workload container.
     """
-    with open("templates/jenkins-config.xml", encoding="utf-8") as jenkins_config_file:
-        container.push(CONFIG_FILE_PATH, jenkins_config_file, user=USER, group=GROUP)
-    with open("templates/logging.properties", encoding="utf-8") as jenkins_logging_config_file:
-        container.push(LOGGING_CONFIG_PATH, jenkins_logging_config_file, user=USER, group=GROUP)
+    _install_config(container, DEFAULT_JENKINS_CONFIG, CONFIG_FILE_PATH)
+
+
+def install_auth_proxy_config(container: ops.Container) -> None:
+    """Install jenkins-config.xml for auth_proxy.
+
+    Args:
+        container: The Jenkins workload container.
+    """
+    _install_config(container, AUTH_PROXY_JENKINS_CONFIG_TEMPLATE, CONFIG_FILE_PATH)
 
 
 def _setup_user_token(container: ops.Container) -> None:
     """Configure admin user API token.
 
     Args:
         container: The Jenkins workload container.
+
+    Raises:
+        JenkinsBootstrapError: if the token can not be setup.
     """
-    client = _get_client(get_admin_credentials(container))
-    token: str = client.generate_new_api_token("juju_api_token")
-    container.push(API_TOKEN_PATH, token, user=USER, group=GROUP)
+    try:
+        client = _get_client(get_admin_credentials(container))
+        token: str = client.generate_new_api_token("juju_api_token")
```

Comment:
I'm wondering if defining "juju_api_token" in a constant might help organize the code a little :)